### ISSUE TITLE
Improve YAML validation by validating the contents of some keys

### ISSUE THREAD

**Issue by iamjackg** [DannyBen/bashly#172]
Hi Danny!

I think it would be very helpful to have some minimal checks on the actual key contents in the configuration.

As an example use case, somebody recently added a two-letter short flag to a command that also had a long-format flag (e.g. `short: -ab`), and they were surprised to discover later on that it didn't work.

Maybe we could add some basic regex checks to `bashly validate` for a few select values that have a known format?

**Comment by DannyBen** [comment]
Yeah, I don't see why not - as long as we keep things simple.

Other than validating that a short flag is a one alphanumeric character, what else would you consider testing?

**Comment by iamjackg** [comment]
Honestly it looks like `long` and `short` are the only two items where there's an implicit format expectation, so maybe just those two? Maybe also making sure that `arg.name` _doesn't_ start with dashes?

**Comment by DannyBen** [comment]
Excellent. Sounds reasonable and logical.
I will implement it sometime in the next few days.

**Comment by DannyBen** [comment]
Added 4 validations:

```ruby
# for arg
refute value['name'].match(/^-/), "#{key}.name must not start with a '-'"

# for flag
assert value['long'].match(/^--[^\s]+$/), "#{key}.long must be in the form of '--name'" if value['long']
assert value['short'].match(/^-[^\s]$/), "#{key}.short must be in the form of '-n'" if value['short']
refute value['arg'].match(/^-/), "#{key}.arg must not start with a '-'" if value['arg']
```

I will leave this PR open for at least 12 hours if any other idea comes up.

**Comment by iamjackg** [comment]
Hmm, this would still let through things like `long: ----lots-of-dashes` and `short: --`. I believe the first one _should_ be okay based on [this regex](https://github.com/DannyBen/bashly/blob/ee76065895a839aadbf7650b3444ea66f18cb2e9/lib/bashly/views/command/normalize_input.erb#L7) (the argument will be called `--lots-of-dashes`), but the second one will not, since neither [this](https://github.com/DannyBen/bashly/blob/ee76065895a839aadbf7650b3444ea66f18cb2e9/lib/bashly/views/command/normalize_input.erb#L10) nor [this](https://github.com/DannyBen/bashly/blob/ee76065895a839aadbf7650b3444ea66f18cb2e9/lib/bashly/views/command/normalize_input.erb#L13) regex allow dashes.

**Comment by DannyBen** [comment]
Well.

1. My usual approach to protections such as this validation case is to help protect against accidental mistakes, not against intentional ones like `----lots-of-dashes`.
2. However, I forgot that I already have some regexes that require a more strict form anyways, so yes - if a stricter form is enforced downstream, it should be protected as such in the validation.

So, I will update the regexes to match the ones in the `normalize_input` bash function. Thanks for pointing that out.

**Comment by DannyBen** [comment]
Updated the flag validations:

```ruby
assert value['long'].match(/^--[a-zA-Z0-9_\-]+$/), "#{key}.long must be in the form of '--name'" if value['long']
assert value['short'].match(/^-[a-zA-Z0-9]$/), "#{key}.short must be in the form of '-n'" if value['short']
```

**Comment by iamjackg** [comment]
Awesome, looks good to me!

**Comment by DannyBen** [comment]
Cool. Merging, and will release tomorrow with this and #171

**Comment by DannyBen** [comment]
This is now implemented in v0.7.3, thanks for the suggestion and assistance.
